{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f60017e9-1e16-4a08-8157-6c6fe32c3a8f",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import scanpy as sc\n",
    "import rawpy, zipfile, tarfile, logging,os\n",
    "import gzip\n",
    "from Bio import SeqIO\n",
    "\n",
    "\n",
    "#folder\n",
    "root_folder = \"C:\\\\Users\\\\Van\\\\Desktop\\\\Python\\\\rnaseq\\\\\"\n",
    "compressed_file =  \"SRR13981697.fastq.gz\" #\"pbmc3k_filtered_gene_bc_matrices.tar.gz\" #\"GSE169048_RAW.tar\"\n",
    "\n",
    "def extract_file(root_folder,compressed_file):\n",
    "\n",
    "    results_file = root_folder + compressed_file\n",
    "    with gzip.open(results_file, \"rt\") as handle:\n",
    "        print(\"opening gzip file\")\n",
    "        for record in SeqIO.parse(handle, \"fastq\"):\n",
    "            print(record.id)\n",
    "                \n",
    "        # pass\n",
    "    new_folder = compressed_file.split(\"_\")[0]\n",
    "    file_extract_folder = tar_file.extractall(root_folder + new_folder)\n",
    "    curr_folder = root_folder + new_folder+\"\\\\\"\n",
    "    \n",
    "    list_iter = 0\n",
    "    isDir = os.path.isdir(curr_folder)\n",
    "\n",
    "    while isDir is True:\n",
    "        dir_list = os.listdir(curr_folder)\n",
    "        for folder in dir_list:\n",
    "            prev_folder = curr_folder\n",
    "            curr_folder = curr_folder+folder+\"\\\\\"\n",
    "            isDir = os.path.isdir(curr_folder)\n",
    "            print(\"Current folder: \",curr_folder, isDir)\n",
    "            list_iter +=1\n",
    "            if isDir is False:\n",
    "                curr_folder = prev_folder\n",
    "                break\n",
    "            \n",
    "    return os.path.dirname(curr_folder)\n",
    "            \n",
    "\n",
    "file_directory = extract_file(root_folder,compressed_file)\n",
    "print(\"Directory:\", file_directory) \n",
    "\n",
    "sc.settings.verbosity = 3\n",
    "sc.settings.set_figure_params(dpi=88, facecolor = \"white\")\n",
    "\n",
    "file_list = os.listdir(file_directory)\n",
    "\n",
    "# for rnaseq_file in file_list:\n",
    "#     print(rnaseq_file)\n",
    "filename = \"\\\\\".join([file_directory,\"genes.tsv\"])\n",
    "df = pd.read_csv(filename)\n",
    "print(\"Columns: \",df.columns, \"\\n\",df.iloc[0,0])\n",
    "\n",
    "data_EYS = df[df.iloc[:,0].str.contains('EYS')]\n",
    "print(\"EYS rows: \", data_EYS)\n",
    "\n",
    "# sc.pl.highest_expr_genes(adata, n_top=20)\n",
    "print('done')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
